src/os/KeyValueStore.cc

```diff
@@ -582,9 +582,13 @@ KeyValueStore::~KeyValueStore()
 
 int KeyValueStore::statfs(struct statfs *buf)
 {
-  if (::statfs(basedir.c_str(), buf) < 0) {
-    int r = -errno;
-    return r;
+  if (g_conf->keyvaluestore_backend != "pluggabledb") {
+    if (::statfs(basedir.c_str(), buf) < 0) {
+      int r = -errno;
+      return r;
+    }
+  } else {
+    backend->db->get_statfs(buf);
   }
```

Comment:
Minor: maybe here can be more friendly, default keyvaluedb will return -EOPNOTSUPP. Keyvaluestore will falldown to use statfs if -EOPNOTSUPP